laws/src/main/scala/cats/laws/discipline/ExhaustiveCheck.scala

```diff
@@ -0,0 +1,55 @@
+package cats
+package laws
+package discipline
+
+/**
+ * An `ExhuastiveCheck[A]` instance can be used similarly to a Scalacheck
+ * `Gen[A]` instance, but differs in that it generates a `Stream` of the entire
+ * domain of values as opposed to generating a random sampling of values.
+ */
+trait ExhaustiveCheck[A] extends Serializable { self =>
+  def allValues: Stream[A]
```

Comment:
Hmm, my thinking is that `Stream` as a type doesn't enforce exhaustiveness, you probably have a practical case for favoring `Stream` over `List`/`Chain` somewhere right?

Reply:
The `Eq` instances that use `ExhaustiveCheck` use `forall`, so my thought was that you wouldn't necessarily need to generate all instances for tests that fail fast. But since this is intended for small domains, there's really no reason to think that `Stream` would be better than `List`. I'm happy to change it if you'd like.